extensions/npm/src/tasks.ts

```diff
@@ -107,8 +108,20 @@ export function isWorkspaceFolder(value: any): value is WorkspaceFolder {
 	return value && typeof value !== 'number';
 }
 
-export function getPackageManager(folder: WorkspaceFolder): string {
-	return workspace.getConfiguration('npm', folder.uri).get<string>('packageManager', 'npm');
+export async function getPackageManager(folder: WorkspaceFolder): Promise<string> {
+	let packageManagerName = workspace.getConfiguration('npm', folder.uri).get<string>('packageManager', 'npm');
+
+	if (packageManagerName === 'auto') {
+		const { name, multiplePMDetected } = await findPreferredPM(folder.uri.fsPath);
+		packageManagerName = name;
+
+		if (multiplePMDetected) {
+			const multiplePMWarning = localize('npm.multiplePMWarning', 'Found multiple lockfiles. Using {0} as the preferred package manager.', packageManagerName);
```

Comment:
Since each `folder` can have multiple PMs, we should include the folder's fspath in this message.